### PR TITLE
delete all versions not present in repository and cleanup version mismatches using schedule

### PR DESCRIPTION
**Pull request by gs-gunjan**
delete all versions not present in repository and cleanup version mismatches using schedule

### CODE REVIEW

File: legend-depot-artifacts-purge/src/main/java/org/finos/legend/depot/store/artifacts/purge/services/ArtifactsPurgeServiceImpl.java [hunk 1]
```diff
@@ -124,15 +150,14 @@ private MetadataEventResponse deleteOldestProjectVersions(ProjectData project, i
                     delete(project.getGroupId(), project.getArtifactId(), versionId.toVersionIdString());
                     versionIds.remove(versionId);
                     project.removeVersion(versionId.toVersionIdString());
-                    response.addMessage(String.format("%s-%s-%s purge", project.getGroupId(), project.getArtifactId(), versionId.toVersionIdString()));
-                    PrometheusMetricsFactory.getInstance().incrementCount(VERSION_PURGE_COUNTER);
```
Comment (gs-gunjan) [inline legend-depot-artifacts-purge/src/main/java/org/finos/legend/depot/store/artifacts/purge/services/ArtifactsPurgeServiceImpl.java:128]: removed the increment in counter here as we already do this when we do delete in line 150

File: legend-depot-artifacts-purge/src/main/java/org/finos/legend/depot/store/artifacts/purge/ArtifactsPurgeModule.java [hunk 2]
```diff
@@ -30,4 +39,13 @@ protected void configure()
 
         bind(ArtifactsPurgeService.class).to(ArtifactsPurgeServiceImpl.class);
     }
+
+    @Provides
+    @Singleton
+    @Named("delete-versions-not-in-repository")
+    boolean initDeleteVersionsNotInRepository(SchedulesFactory schedulesFactory, ArtifactsPurgeService purgeService, ArtifactRepositoryProviderConfiguration configuration)
+    {
+        schedulesFactory.register(DELETE_VERSIONS_NOT_IN_REPO, LocalDateTime.now().plusSeconds(configuration.getVersionsUpdateIntervalInMillis() / 1000), configuration.getFixVersionsMismatchIntervalInMillis(), false,() -> purgeService.deleteVersionsNotInRepository());
```
Comment (elopezcastro) [inline legend-depot-artifacts-purge/src/main/java/org/finos/legend/depot/store/artifacts/purge/ArtifactsPurgeModule.java:48]: we should have a separate config for this

Comment (gs-gunjan) [reply legend-depot-artifacts-purge/src/main/java/org/finos/legend/depot/store/artifacts/purge/ArtifactsPurgeModule.java]: added separate config attribute

File: legend-depot-artifacts-purge/src/main/java/org/finos/legend/depot/store/artifacts/purge/services/ArtifactsPurgeServiceImpl.java [hunk 3]
```diff
@@ -104,12 +108,34 @@ public void delete(String groupId, String artifactId, String versionId)
...
-    public MetadataEventResponse deleteOldestProjectVersions(String groupId, String artifactId, int versionsToKeep)
+    public MetadataEventResponse evictVersionsNotInRepository()
     {
-        return deleteOldestProjectVersions(getProject(groupId, artifactId),versionsToKeep);
+        List<VersionMismatch> versionMismatch = repository.findVersionsMismatches();
+        return deleteVersionMismatches(versionMismatch);
     }
 
-    private MetadataEventResponse deleteOldestProjectVersions(ProjectData project, int versionsToKeep)
+    public MetadataEventResponse deleteVersionMismatches(List<VersionMismatch> versionMismatch)
```
[the remark this answers is not in the document]
Comment (gs-gunjan) [reply legend-depot-artifacts-purge/src/main/java/org/finos/legend/depot/store/artifacts/purge/services/ArtifactsPurgeServiceImpl.java]: renamed this to make it more clear

**Review by elopezcastro — APPROVED** [review]

**Pull request merged**